node/src/components/linear_chain_sync.rs

```diff
@@ -622,3 +694,45 @@ where
             move || Event::GetBlockHeightResult(block_height, BlockByHeightResult::Absent(cloned)),
         )
 }
+
+/// Returns key in the database, under which the LinearChainSync's state is stored.
+fn state_key(chainspec: &Chainspec) -> Vec<u8> {
+    chainspec.network_config.name.clone().into()
+}
+
+/// Deserialized vector of bytes into `LinearChainSync::State`.
+/// Panics on deserialization errors.
+fn deserialize_state(serialized_state: &[u8]) -> Option<State> {
+    bincode::deserialize(&serialized_state).unwrap_or_else(|error| {
+        panic!(
+            "could not deserialize state from storage, error {:?}",
+            error
+        )
+    })
+}
+
+/// Reads the `LinearChainSync's` state from storage, if any.
+/// Panics on deserialization errors.
+pub(crate) fn read_init_state(
+    storage: &Storage,
```

Comment:
Passing a ref to one component into another component's function is very much against the spirit of the reactor pattern we're using.  I'm OK with this as a stopgap measure, but we should see this as a high priority item to be refactored.

Reply:
I thought about this but this isn't a reactor yet, it's still an initialization/setup phase and we're passing reference, not moving the value.

Reply:
Does this sound good to you? cc @marc-casperlabs